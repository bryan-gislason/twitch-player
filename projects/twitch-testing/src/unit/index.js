import { ThemeProvider } from "@twitch-player/themes/dist/context";
import { light as lightTheme } from "@twitch-player/themes/dist/themes";
import { OverridesProvider } from "@twitch-player/ui/dist/context";
import React from "react";
import { act } from "react-dom/test-utils";
import { MemoryRouter as Router } from "react-router";
import { fireEvent, render } from "react-testing-library";

const customRender = (node, ...options) => {
  return render(
    <ThemeProvider theme={lightTheme}>
      <OverridesProvider>
        <Router>{node}</Router>
      </OverridesProvider>
    </ThemeProvider>,
    ...options
  );
};

fireEvent["press"] = (node, init) => {
  fireEvent.mouseDown(node, init);
  fireEvent.mouseUp(node, init);
};

fireEvent["resize"] = (node, init) => {
  const { target: { left = 0, top = 0, width = 0, height = 0 } = {} } = init;

  Object.defineProperties(window.HTMLElement.prototype, {
    offsetLeft: { get: () => left },
    offsetTop: { get: () => top },
    offsetWidth: { get: () => width },
    offsetHeight: { get: () => height },
  });

  const eventInit = { bubbles: false, cancelable: false };
  const event = new Event(node, eventInit);

  act(() => {
    fireEvent(node, event);
    // eslint-disable-next-line no-undef
    jest.runAllTimers();
  });
};

export * from "react-testing-library";

export { customRender as render };
