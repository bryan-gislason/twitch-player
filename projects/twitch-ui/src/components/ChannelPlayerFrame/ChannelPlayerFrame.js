import PropTypes from "prop-types";
import React, { memo } from "react";

const ChannelPlayerFrame = ({ name, testID = "channel-player" }) => {
  return (
    <iframe
      style={styles.root}
      src={`https://player.twitch.tv/?channel=${name}`}
      allowFullScreen={true}
      data-testid={testID}
    />
  );
};

ChannelPlayerFrame.propTypes = {
  name: PropTypes.string.isRequired,
  testID: PropTypes.string,
};

const styles = {
  root: {
    border: 0,
    flex: 1,
  },
};

export default memo(ChannelPlayerFrame);
