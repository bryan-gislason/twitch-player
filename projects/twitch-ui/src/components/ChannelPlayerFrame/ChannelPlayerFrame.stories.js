import { storiesOf } from "@storybook/react";
import React from "react";
import ChannelPlayerFrame from ".";

storiesOf("Components|ChannelPlayerFrame", module).add("default", () => (
  <ChannelPlayerFrame name="twitchpresents" />
));
