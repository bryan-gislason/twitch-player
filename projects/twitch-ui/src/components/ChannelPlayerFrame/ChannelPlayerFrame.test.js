import { render } from "@twitch-player/testing/dist/unit";
import React from "react";
import ChannelPlayerFrame from "./ChannelPlayerFrame";

describe("ChannelPlayerFrame", () => {
  let instance;

  beforeEach(() => {
    instance = render(<ChannelPlayerFrame name="twitchpresents" />);
  });

  it("renders a channel player", () => {
    expect(instance.getByTestId("channel-player")).toMatchSnapshot();
  });
});
