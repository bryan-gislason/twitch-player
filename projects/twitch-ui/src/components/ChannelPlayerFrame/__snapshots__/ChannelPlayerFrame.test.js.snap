// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelPlayerFrame renders a channel player 1`] = `
<iframe
  allowfullscreen=""
  data-testid="channel-player"
  src="https://player.twitch.tv/?channel=twitchpresents"
  style="border: 0px; flex: 1;"
/>
`;
